config/jobs/kubernetes/sig-node/sig-node-presubmit.yaml

```diff
@@ -4276,7 +4278,10 @@ presubmits:
     skip_branches:
     - release-\d+\.\d+  # per-release image
     always_run: false
-    run_if_changed: /dra/|/dynamicresources/|/resourceclaim/|/resourceclass/|/podscheduling/|/resourceclaimtemplate/|/dynamic-resource-allocation/|/pkg/apis/resource/|/api/resource/|/test/e2e_node/dra_test.go
+    # Automatically testing with one container runtime is sufficient.
+    # containerd was picked because it's apparently more common.
```

Comment:
@bart0sh: does choosing containerd make sense?

Which runtime is more stable in your experience?

Reply:
cri-o just a little bit more stable lately, but can and most probably will change if we drop one of the runtimes as you're suggesting.

Reply:
@pohly 
> Automatically testing with one container runtime is sufficient.

Sufficient for what?

Reply:
Can we keep this job as is and stop automatically triggering `pull-kubernetes-node-e2e-crio-cgrpv1-dra` instead?

Reply:
> cri-o just a little bit more stable lately, but it can and most probably will change if we drop one of the runtimes as you're suggesting.

Why would dropping testing of one runtime change how stable the other runtime is?

> Sufficient for what?

Sufficient for knowing that kubelet's interaction with a runtime isn't completely broken.

> Can we keep this job as is and stop automatically triggering pull-kubernetes-node-e2e-crio-cgrpv1-dra instead?

Sure, if that one was more stable in the past. The goal is to avoid flakes that aren't caused by our code changes, so whatever runtime is more stable is better.

Reply:
Changed.

Reply:
Sorry for the confusion. I proposed to keep both containerd and crio cgroupv2 jobs automatially triggered and disable automatic triggering for cgroupv1 job.

Reply:
Have we ever had issues found during pre-submit that only showed up with only one of the two runtimes?

If no, then testing with both during pre-submit was unnecessary and probably is also unnecessary going forward.

Reply:
I don't remember if we had any issues only for one runtime, but even if we didn't, it doesn't mean that this is unnecessary, at least from my point of view.

Reply:
One more point is that containerd and crio jobs are using different distros. When I added those jobs my idea was to cover all available runtimes and distros. Disabling this job disables testing on cos and ubuntu as well as on containerd.

Reply:
> But we don't need that level of confidence for every single PR that we merge.

What's wrong with having an ability to see a failure before merging the code causing it? And why we don't need to test it for containerd, cos and ubuntu, but need it for cri-o and fedora? Where is a border line between pr and ci jobs in this sense?

Reply:
For pre-submit: what's wrong is that this causes real, significant costs for the projects without being justified by the benefit that it brings. We cannot afford to run everything because "there might be failures".

For periodic: here it is more reasonable to expand the scope of platform coverage because we control how often we test. But also here the question arises whether we expect problems on one platform that don't occur on another and whether it should be Kubernetes that tests that or the downstream consumer of Kubernetes. Even cloud providers are expected to run their own testing of Kubernetes these days, not the Kubernetes project itself.

Reply:
It's hard to argue about costs without having any information about it. However,  having one more job triggered on the changes to a tiny subset of codebase wouldn't make a lot of difference in my opinion. So, I'm still for keeping both jobs active.

Let's see what other reviewers say about it.

Reply:
> It's hard to argue about costs without having any information about it

Then just trust Ben when he says that automatically triggered pre-submit jobs shouldn't exist at all or only under very special circumstances. He is SIG testing and infra lead, he knows about the costs.

> having one more job triggered on the changes to a tiny subset of codebase

I think you haven't seen all PRs where this got triggered. I got asked several times about them in the context of PRs where they shouldn't have been started.